pkg/crc/preflight/preflight_linux.go

```diff
@@ -146,21 +151,34 @@ func fixVsock() error {
 	if err != nil {
 		return err
 	}
-	_, _, err = crcos.RunWithPrivilege("modprobe vhost_vsock", "modprobe", "vhost_vsock")
+
+	udevRule := `KERNEL=="vsock", MODE="0060", OWNER="root", GROUP="libvirt"`
+
+	path, err := exec.LookPath("sudo")
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to find sudo on path: %w", err)
 	}
-	_, _, err = crcos.RunWithPrivilege("chown /dev/vsock", "chown", "root:libvirt", "/dev/vsock")
+	cmd := exec.Command(path, "tee", vsockUdevRulesPath)
+	cmd.Stdin = strings.NewReader(udevRule)
+	logging.Infof("Will use root access: creating %s", vsockUdevRulesPath)
+	out, err := cmd.CombinedOutput()
 	if err != nil {
+		logging.Debug("Failed to create vsock udev rule: ", out)
 		return err
 	}
-	_, _, err = crcos.RunWithPrivilege("chmod /dev/vsock", "chmod", "g+rw", "/dev/vsock")
+
+	_, _, err = crcos.RunWithPrivilege("modprobe vhost_vsock", "modprobe", "vhost_vsock")
```

Comment:
Don't we need an additional rule (or expand the current one) to make sure the module is loaded? If you reboot the host, I don't think `vhost_vsock` will be loaded by default? 
Rule would probably look like this: (I did not test it, nor checked the pci id is correct)
```
ACTION=="add", SUBSYSTEM=="pci", ATTR{vendor}=="0x1af4", ATTR{device}=="0x1012", RUN{builtin}+="kmod load vhost_vsock"
```

Reply:
i did not check for the module being loaded on reboot, only verified the permissions were correct after a reboot, will test it out and update

Reply:
> Don't we need an additional rule (or expand the current one) to make sure the module is loaded? If you reboot the host, I don't think `vhost_vsock` will be loaded by default?
> Rule would probably look like this: (I did not test it, nor checked the pci id is correct)
> 
> ```
> ACTION=="add", SUBSYSTEM=="pci", ATTR{vendor}=="0x1af4", ATTR{device}=="0x1012", RUN{builtin}+="kmod load vhost_vsock"
> ```

Is the rule supposed to match a parent device of vsock? for /dev/vsock it doesn't have the vendor and device keys..
```
❯ udevadm info -a -n /dev/vsock                    

Udevadm info starts with the device specified by the devpath and then
walks up the chain of parent devices. It prints for every device
found, all possible attributes in the udev rules key format.
A rule to match, can be composed by the attributes of the device
and the attributes from one single parent device.

  looking at device '/devices/virtual/misc/vsock':
    KERNEL=="vsock"
    SUBSYSTEM=="misc"
    DRIVER==""
    ATTR{power/control}=="auto"
    ATTR{power/runtime_active_time}=="0"
    ATTR{power/runtime_status}=="unsupported"
    ATTR{power/runtime_suspended_time}=="0"
```

Alternatively we could create config file `/etc/modules-load.d/vhost_vsock.conf` for automatic loading of `hvsock_vsock` module?

Reply:
Yeah, actually this `SUBSYSTEM=="pci", ATTR{vendor}=="0x1af4", ATTR{device}=="0x1012"` would work inside the VM where the virtio device is a PCI device. On the host, I don't know, and it's true there is no clear trigger we can hook up to in order to load the module, so yes, probably best to just drop a file in `/etc/modules-load.d`